tests/test_AggregateActionxData.cpp

```diff
@@ -0,0 +1,583 @@
+#define BOOST_TEST_MODULE UDQ-ACTIONX_Data
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Well/Well2.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+
+#include <opm/output/eclipse/AggregateUDQData.hpp>
+#include <opm/output/eclipse/AggregateActionxData.hpp>
+#include <opm/output/eclipse/WriteRestartHelpers.hpp>
+
+#include <opm/output/eclipse/InteHEAD.hpp>
+#include <opm/output/eclipse/VectorItems/intehead.hpp>
+#include <opm/output/eclipse/DoubHEAD.hpp>
+
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQInput.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQActive.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQConfig.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQParams.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Action/ActionX.hpp>
+
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+#include <opm/parser/eclipse/Units/Units.hpp>
+
+#include <opm/io/eclipse/OutputStream.hpp>
+
+#include <stdexcept>
+#include <utility>
+#include <exception>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace {
+    
+    Opm::Deck first_sim(std::string fname) {
+        return Opm::Parser{}.parseFile(fname);        
+    } 
+}
+/*
+Opm::SummaryState sum_state_TEST1_FLOW()
+    {
+        auto state = Opm::SummaryState{std::chrono::system_clock::now()};
+        state.update_well_var("OPU01", "WUPR3", 4);
+        state.update_well_var("OPU02", "WUPR3", 3);
+        state.update_well_var("OPL01", "WUPR3", 1);
+        state.update_well_var("OPL02", "WUPR3", 2);
+        
+        state.update("FMWPR", 4);
+                
+
+        return state;
+}
+*/
+Opm::SummaryState sum_state_TEST1()
+    {
+        auto state = Opm::SummaryState{std::chrono::system_clock::now()};
+        state.update_well_var("OPU01", "WWPR", 21.);
+        state.update_well_var("OPU02", "WWPR", 22.);
+        state.update_well_var("OPL01", "WWPR", 23.);
+        state.update_well_var("OPL02", "WWPR", 24.);
+
+        state.update_well_var("OPU01", "WGPR", 230.);
+        state.update_well_var("OPU02", "WGPR", 231.);
+        state.update_well_var("OPL01", "WGPR", 232.);
+        state.update_well_var("OPL02", "WGPR", 233.);
+
+        state.update_group_var("UPPER", "GWPR", 36.);
+        state.update_group_var("LOWER", "GWPR", 37.);
+        state.update_group_var("TEST",  "GWPR", 73.);
+        
+        state.update_group_var("UPPER", "GGPR", 460.);
+        state.update_group_var("LOWER", "GGPR", 461.);
+        state.update_group_var("TEST",  "GGPR", 821.);
+
+        
```

Comment:
Adding vertical space is a good way to indicate codeline which belong together, and where there is a change of context - here the vertical spacing seems a bit random?